File: core/jobs/src/main/java/com/cognifide/aet/job/common/collectors/accessibility/AccessibilityIssue.java
/**
 * AET
 *
 * Copyright (C) 2013 Cognifide Limited
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License
 * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing permissions and limitations under
 * the License.
 */
package com.cognifide.aet.job.common.collectors.accessibility;

import com.cognifide.aet.job.common.Excludable;
import java.io.Serializable;

public class AccessibilityIssue implements Serializable, Excludable {

  private static final long serialVersionUID = -53665467524179701L;

  private final IssueType type;

  private final String message;

  private final String code;

  private final String elementString;

  private final String elementStringAbbreviated;

  private int lineNumber;

  private int columnNumber;

  /**
   * If issue is excluded from results, i.e. is listed but not taken into account when computed
   * results.
   */
  private boolean excluded;

  public AccessibilityIssue(IssueType type, String message, String code, String elementString,
      String elementStringAbbreviated) {
    this.message = message;
    this.code = code;
    this.type = type;
    this.elementString = elementString;
    this.elementStringAbbreviated = elementStringAbbreviated;
  }

  public String getMessage() {
    return message;
  }

  public String getCode() {
    return code;
  }

  public IssueType getType() {
    return type;
  }

  public String getElementString() {
    return elementString;
  }

  public String getElementStringAbbreviated() {
    return elementStringAbbreviated;
  }

  public int getLineNumber() {
    return lineNumber;
  }

  public void setLineNumber(int lineNumber) {
    this.lineNumber = lineNumber;
  }

  public int getColumnNumber() {
    return columnNumber;
  }

  public void setColumnNumber(int columnNumber) {
    this.columnNumber = columnNumber;
  }

  @Override
  public boolean isExcluded() {
    return excluded;
  }

  @Override
  public void exclude() {
    this.excluded = true;
  }

  public enum IssueType {
    ERROR("1"),
    WARN("2"),
    NOTICE("3"),
    UNKNOWN("");

    private String value;

    IssueType(String value) {
      this.value = value;
    }

    public static IssueType byValue(String value) {
      for (IssueType issueType : IssueType.values()) {
        if (issueType.value.equalsIgnoreCase(value)) {
          return issueType;
        }
      }
      return UNKNOWN;
    }
  }
}

File: core/accessibility-report/src/main/kotlin/com/cognifide/aet/accessibility/report/models/AccessibilityIssue.kt
/*
 * AET
 *
 * Copyright (C) 2020 Cognifide Limited
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License
 * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing permissions and limitations under
 * the License.
 */
package com.cognifide.aet.accessibility.report.models

import java.io.Serializable
import java.util.Objects

class AccessibilityIssue(
    val type: IssueType,
    val message: String,
    val code: String,
    val elementString: String,
    val elementStringAbbreviated: String) : Serializable {

  var lineNumber = 0
  var columnNumber = 0
  var isExcluded = false
    private set
  var url: String = ""

  fun getAccessibilityCode(): AccessibilityCode = AccessibilityCode(code)

  override fun equals(other: Any?): Boolean {
    if (this === other) {
      return true
    }
    if (other == null || javaClass != other.javaClass) {
      return false
    }

    val that = other as AccessibilityIssue
    return (lineNumber == that.lineNumber
        && columnNumber == that.columnNumber
        && isExcluded == that.isExcluded
        && type == that.type
        && message == that.message
        && code == that.code
        && elementString == that.elementString
        && elementStringAbbreviated == that.elementStringAbbreviated
        && url == that.url)
  }

  override fun hashCode(): Int {
    return Objects.hash(
        type,
        message,
        code,
        elementString,
        elementStringAbbreviated,
        lineNumber,
        columnNumber,
        isExcluded,
        url)
  }

  enum class IssueType {
    ERROR, WARN, NOTICE, UNKNOWN;
  }

  companion object {
    private const val serialVersionUID = -53665467524179701L
  }
}
